pallets/broadcast/src/lib.rs

```diff
@@ -61,18 +62,18 @@ pub mod pallet {
 	pub(super) type IncrementalId<T: Config> = StorageValue<_, IncrementalIdType, ValueQuery>;
 
 	#[pallet::storage]
+	#[pallet::whitelist_storage]
 	/// Execution context to figure out where the trade is originated from
 	#[pallet::getter(fn execution_context)]
 	pub(super) type ExecutionContext<T: Config> = StorageValue<_, ExecutionIdStack, ValueQuery>;
 
-	/// To handle the overflow of increasing the execution context.
-	/// After the stack is full, we start to increase the overflow count,
-	/// so we how many times we can ignore the removal from the context.
-	#[pallet::storage]
-	pub(super) type OverflowCount<T: Config> = StorageValue<_, u32, ValueQuery>;
-
 	#[pallet::error]
-	pub enum Error<T> {}
+	pub enum Error<T> {
+		///The execution context has reached its maximum size
+		ExecutionContextFull,
+		///Cannot remove from the execution context as it is empty
```

Comment:
```suggestion
		///Execution context call stack is empty
```

Reply:
This is probably bigger issue than overflow which could happen... This can happen only if you fuk up the impl. Maybe we should do something extra here???

Reply:
Before I added debug_assert! for us, so if we fuk up the impls then the developers know it, if tested. But then the fuzzer started failing on these as it just called this function, triggering the debug assert

Reply:
We could add some logging, so we get notified better if this happens